presto-main/src/test/java/com/facebook/presto/execution/buffer/TestClientBuffer.java

```diff
@@ -361,6 +361,48 @@ public void testReferenceCount()
         assertBufferDestroyed(buffer, 1);
     }
 
+    @Test
+    public void testProcessReadLockHolderAssertionsFireInTest()
+    {
+        ClientBuffer buffer = new ClientBuffer(TASK_INSTANCE_ID, BUFFER_ID, NOOP_RELEASE_LISTENER);
+        try {
+            ListenableFuture<BufferResult> pendingRead = buffer.getPages(0, DataSize.succinctBytes(1));
+            synchronized (buffer) {
+                addPage(buffer, createPage(0));
+            }
+            fail("Expected AssertionError to be thrown, are assertions enabled in your testing environment?");
+            assertTrue(getFuture(pendingRead, NO_WAIT).isEmpty(), "Code should not reach here");
+        }
+        catch (AssertionError ae) {
+            assertEquals(ae.getMessage(), "Can not process pending read while holding a lock on this");
+        }
+        finally {
+            buffer.destroy();
+        }
+    }
+
+    @Test
+    public void testGetPagesWithSupplierLockHolderAssertionsFireInTest()
+    {
+        ClientBuffer buffer = new ClientBuffer(TASK_INSTANCE_ID, BUFFER_ID, NOOP_RELEASE_LISTENER);
+        TestingPagesSupplier supplier = new TestingPagesSupplier();
+        supplier.addPage(createPage(0));
+        try {
+            ListenableFuture<BufferResult> result;
+            synchronized (buffer) {
+                result = buffer.getPages(0, sizeOfPages(1), Optional.of(supplier));
+            }
+            fail("Expected AssertionError to be thrown, are assertions enabled in your testing environment?");
+            assertTrue(getFuture(result, NO_WAIT).isEmpty(), "Code should not reach here");
```

Comment:
I don't get why there should be something after fail?

Reply:
Makes the reference used to satisfy some pedantic checkstyles but also the potential JIT DCE “optimizations”. Also, IDEs. Should be safe to omit, but I didn’t like how IntelliJ was judging me without it.